arangod/ClusterEngine/ClusterRestWalHandler.cpp

```diff
@@ -72,6 +72,27 @@ RestStatus ClusterRestWalHandler::execute() {
       properties();
       return RestStatus::DONE;
     }
+  } else if (operation == "wait_for_estimator_sync") {
+#ifndef ARANGODB_ENABLE_MAINTAINER_MODE
+    if (!ExecContext::current().isSuperuser()) {
+      generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_HTTP_FORBIDDEN,
+                    "system level access is needed for this API");
+      return RestStatus::DONE;
+    }
+#else
+    if (!ExecContext::current().isAdminUser()) {
+      generateError(rest::ResponseCode::FORBIDDEN, TRI_ERROR_HTTP_FORBIDDEN,
+                    "you need admin rights to produce a cluster info dump");
+      return RestStatus::DONE;
+    }
+#endif
+    server()
+        .getFeature<EngineSelectorFeature>()
+        .engine()
+        .waitForEstimatorSync();
+    generateResult(rest::ResponseCode::OK,
+                   arangodb::velocypack::Slice::emptyObjectSlice());
+    return RestStatus::DONE;
```

Comment:
Do we even want this route to be available outside of maintainer mode? If yes, why do we have different permission checks though?

Reply:
Do we care about this test in prod environments? Should we instead just skip the test for non maintainer builds?

I am not really a fan of having functionality in production builds that are only relevant for tests.